src/main/java/org/wildfly/extension/elytron/SSLDefinitions.java

```diff
@@ -269,6 +282,48 @@ static ResourceDefinition getKeyManagerDefinition() {
 
     }
 
+    private static void wrapKeyManagersByAliasSelector(KeyManager[] keyManagers, String alias) {
+        for (int i = 0; i < keyManagers.length; i++) {
+            if (keyManagers[i] instanceof X509ExtendedKeyManager) {
+
+                final X509ExtendedKeyManager delegate = (X509ExtendedKeyManager) keyManagers[i];
+
+                keyManagers[i] = new X509ExtendedKeyManager() {
+
+                    @Override
+                    public String chooseClientAlias(String[] keyType, Principal[] issuers, Socket socket) {
+                        return alias;
```

Comment:
This is the kind of code we are trying to avoid, this bypasses the underlying KeyManager having an opportunity to decide if the alias is suitable for use with the listed key types.